yt/geometry/coordinates/spherical_coordinates.py

```diff
@@ -277,27 +285,105 @@ def image_axis_name(self):
     def period(self):
         return self.ds.domain_width
 
+    def _get_poloidal_bounds(self):
+        ri = self.axis_id["r"]
+        ti = self.axis_id["theta"]
+        rmin = self.ds.domain_left_edge[ri]
+        rmax = self.ds.domain_right_edge[ri]
+        thetamin = self.ds.domain_left_edge[ti]
+        thetamax = self.ds.domain_right_edge[ti]
+        corners = [
+            (rmin, thetamin),
+            (rmin, thetamax),
+            (rmax, thetamin),
+            (rmax, thetamax),
+        ]
+
+        def to_poloidal_plane(r, theta):
+            # take a r, theta position and return
+            # cylindrical R, z coordinates
+            R = r * np.sin(theta)
+            z = r * np.cos(theta)
+            return R, z
+
+        cylindrical_corner_coords = [to_poloidal_plane(*corner) for corner in corners]
+
+        thetamin = thetamin.d
+        thetamax = thetamax.d
+
+        Rmin = min(R for R, z in cylindrical_corner_coords)
+
+        if thetamin <= np.pi / 2 <= thetamax:
+            Rmax = rmax
+        else:
+            Rmax = max(R for R, z in cylindrical_corner_coords)
+
+        zmin = min(z for R, z in cylindrical_corner_coords)
```

Comment:
Would it be possible to consolidate this and the `Rmin` calculation into a set of by-axis min/max numpy call?

Reply:
hum. Actually, and this may be fatigue speaking but it seems to me that turning these into numpy operations would make the code a lot harder to read and maintain for a very negligible performance gain.

Reply:
Sounds good!